union.go

```diff
@@ -0,0 +1,58 @@
+package rwvfs
+
+import (
+	"io"
+	pathpkg "path"
+
+	"golang.org/x/tools/godoc/vfs"
+)
+
+// Union returns a new FileSystem which is the union of the provided file systems.
+func Union(fileSystems ...FileSystem) FileSystem {
+	if len(fileSystems) == 0 {
+		return Map(nil)
+	}
+	if len(fileSystems) == 1 {
+		return fileSystems[0]
+	}
+
+	ns := vfs.NameSpace{}
+	for _, fs := range fileSystems {
+		ns.Bind("/", fs, "/", vfs.BindAfter)
+	}
+	return &unionFs{
+		NameSpace:   ns,
+		fileSystems: fileSystems,
+	}
+}
+
+type unionFs struct {
+	vfs.NameSpace
+	fileSystems []FileSystem
+}
```

Comment:
Add a `(unionFS).String() string` method to unionFS, or else it'll fall back to `vfs.Namespace`'s String method (which will be confusing).